pytorch_lightning/trainer/connectors/debugging_connector.py

```diff
@@ -31,16 +31,32 @@ def on_init_start(
             overfit_batches,
             fast_dev_run
     ):
+        if not isinstance(fast_dev_run, (bool, int)):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a valid configuration.'
+                ' It should be either a bool or an int >= 0'
+            )
+
+        if isinstance(fast_dev_run, int) and (fast_dev_run < 0):
+            raise MisconfigurationException(
+                f'fast_dev_run={fast_dev_run} is not a'
+                ' valid configuration. It should be >= 0.'
+            )
 
         self.trainer.fast_dev_run = fast_dev_run
-        if self.trainer.fast_dev_run:
-            limit_train_batches = 1
-            limit_val_batches = 1
-            limit_test_batches = 1
+
+        if fast_dev_run is True:
+            fast_dev_run = 1
+
+        if fast_dev_run:
```

Comment:
Not sure: If `fast_dev_run` > 1, I think `fast_dev_run ` it should be set to False, so logging works properly.

Reply:
@tchaton you mean if `fast_dev_run > 1` it should follow `log_every_n_steps`??

Reply:
Yes, if `fast_dev_run > 1`, I would expect him to follow `log_every_n_steps `. Do you agree ?

Reply:
is it good now?